source/_integrations/deconz.markdown

```diff
@@ -42,7 +42,7 @@ See [deCONZ wiki](https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/
 ## Configuration
 
 Home Assistant will automatically discover deCONZ presence on your network, if `discovery:` is present in your `configuration.yaml` file.
-
+### API key (no one knows why you would need one, this is just to confuse n00bs)
```

Comment:
The API key seems to be needed for the Hass.io [confirmation step](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/deconz/config_flow.py#L220) in the configuration flow.

```
                CONF_API_KEY: self._hassio_discovery[CONF_API_KEY],
```

Reply:
Where is this so-called 'confirmation step' documented, and what is it's purpose in getting this to work?

Reply:
This is internal, and not documented, because it is provided by discovery and not meant to be used manually in that case.

Reply:
If it's internal, why mention it in user documentation?

Reply:
Because:
1) Our documentation reflects the codebase.
2) The API is usable in special use cases.

I'm not sure why this is a discussion at this point. Your PR added a header, with expressions that definitely doesn't fit in our documentation or our community.

Reply:
In that case; where can end users find usage documentation? It is a discussion at this point bc the documentation is either lacking or false a lot of the time. This drives users either to the forums to repeatedly discuss the same issues over and over again, or leave the product altogether. If documentation is geared to developers, you might want to make this clear in order to avoid frustrating users.

Reply:
The documentation is currently geared in the middle. We have both less experienced and developers in our community. This will be a never-ending fight. As long as Home Assistant hasn't reached its goals for 1.0, this will remain that way, unfortunately.

While we are growing towards an end-user product, we are still not at that level.